pallets/subtensor/src/swap.rs

```diff
@@ -0,0 +1,422 @@
+    ///
+    /// # Returns
+    ///
+    /// * `Result<(), Error<T>>` - The result of the operation.
+    pub fn swap_owner(
+        old_hotkey: &T::AccountId,
+        new_hotkey: &T::AccountId,
+        coldkey: &T::AccountId,
+        weight: &mut Weight,
+    ) -> DispatchResult {
+        Owner::<T>::remove(old_hotkey);
+        Owner::<T>::insert(new_hotkey, coldkey.clone());
+        weight.saturating_accrue(T::DbWeight::get().writes(2));
+        Ok(())
+    }
+
+    /// Swaps the total stake of the hotkey.
+    ///
+    /// # Arguments
+    ///
+    /// * `old_hotkey` - The old hotkey.
+    /// * `new_hotkey` - The new hotkey.
+    /// * `weight` - The weight of the transaction.
+    ///
+    /// # Returns
+    ///
+    /// * `Result<(), Error<T>>` - The result of the operation.
+    pub fn swap_total_hotkey_stake(
+        old_hotkey: &T::AccountId,
+        new_hotkey: &T::AccountId,
+        weight: &mut Weight,
+    ) -> DispatchResult {
+        if let Ok(total_hotkey_stake) = TotalHotkeyStake::<T>::try_get(old_hotkey) {
+            TotalHotkeyStake::<T>::remove(old_hotkey);
+            TotalHotkeyStake::<T>::insert(new_hotkey, total_hotkey_stake);
+            weight.saturating_accrue(T::DbWeight::get().writes(2));
+        }
+        Ok(())
+    }
+
+    /// Swaps the delegates of the hotkey.
+    ///
+    /// # Arguments
+    ///
+    /// * `old_hotkey` - The old hotkey.
+    /// * `new_hotkey` - The new hotkey.
+    /// * `weight` - The weight of the transaction.
+    ///
+    /// # Returns
+    ///
+    /// * `Result<(), Error<T>>` - The result of the operation.
+    pub fn swap_delegates(
+        old_hotkey: &T::AccountId,
+        new_hotkey: &T::AccountId,
+        weight: &mut Weight,
+    ) -> DispatchResult {
+        if let Ok(delegate_take) = Delegates::<T>::try_get(old_hotkey) {
+            Delegates::<T>::remove(old_hotkey);
+            Delegates::<T>::insert(new_hotkey, delegate_take);
+            weight.saturating_accrue(T::DbWeight::get().writes(2));
+        }
+        Ok(())
+    }
+
+    /// Swaps the stake of the hotkey.
+    ///
+    /// # Arguments
+    ///
+    /// * `old_hotkey` - The old hotkey.
+    /// * `new_hotkey` - The new hotkey.
+    /// * `weight` - The weight of the transaction.
+    ///
+    /// # Returns
+    ///
+    /// * `Result<(), Error<T>>` - The result of the operation.
+    pub fn swap_stake(
+        old_hotkey: &T::AccountId,
+        new_hotkey: &T::AccountId,
+        weight: &mut Weight,
+    ) -> DispatchResult {
+        let mut writes = 0;
+        let stakes: Vec<(T::AccountId, u64)> = Stake::<T>::iter_prefix(old_hotkey).collect();
+        let stake_count = stakes.len() as u32;
+        for (coldkey, stake_amount) in stakes {
+            Stake::<T>::insert(new_hotkey, &coldkey, stake_amount);
+            let _ = Stake::<T>::clear_prefix(old_hotkey, stake_count, None);
+            writes += 2; // One write for insert and one for remove
+        }
```

Comment:
Have we checked how many keys will be touched here? Could be a very expensive operation